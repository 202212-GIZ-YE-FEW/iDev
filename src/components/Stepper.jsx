import clsx from "clsx";
import { withTranslation } from "next-i18next";
import { React, useState } from "react";

import PageIntro from "./PageIntro";
import Button from "./ui/Button";

function Stepper(props) {
    const { t, steps, currentStep } = props;
    const [current, setCurrent] = useState(
        currentStep >= 0 && currentStep < steps.length ? currentStep : 0
    );
    const nextStep = () => {
        if (current < steps.length - 1) {
            setCurrent(current + 1);
        }
    };

    const prevStep = () => {
        if (current > 0) {
            setCurrent(current - 1);
        }
    };

    return (
        <>
            {steps.map((step, index) => {
                return (
                    <>
                        <div
                            key={index}
                            className={`${
                                current === index ? "flex flex-col" : "hidden"
                            }`}
                        >
                            <PageIntro
                                title={t(`${step.pageTitle}`)}
                                subtitle={t(`${step.pageSubtitle}`)}
                            />
                            <div
                                className={clsx("w-full mx-auto py-5", {
                                    "bg-light-white flex flex-col justify-between max-w-3xl h-[30rem] px-8 rounded-md drop-shadow-lg":
                                        index !== 5,
                                })}
                            >
                                <div className='w-full mb-4'>
                                    <p className='text-3xl font-normal text-start'>
                                        {step.title}
                                    </p>
                                    {step.content}
                                </div>
                                <div className='flex justify-center gap-4 mb-5'>
                                    {current > 0 && (
                                        <Button
                                            content={t("previous")}
                                            onClick={prevStep}
                                            textTransform='uppercase'
                                            filled='true'
                                            size='large'
                                            fontSize='text-lg md:text-xl lg:text-2xl'
                                            radius='md'
                                        />
                                    )}
                                    {current < steps.length - 1 && (
                                        <Button
                                            content={t("next")}
                                            onClick={nextStep}
                                            textTransform='uppercase'
                                            filled='true'
                                            size='large'
                                            fontSize='text-lg md:text-xl lg:text-2xl'
                                            radius='md'
                                        />
                                    )}
                                </div>
                            </div>
                        </div>
                    </>
                );
            })}
        </>
    );
}

export default withTranslation("appointment")(Stepper);
