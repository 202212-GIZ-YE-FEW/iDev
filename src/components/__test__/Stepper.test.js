import { fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";

import Stepper from "../Stepper";

const steps = [
    {
        title: "Step 1",
        content: "This is the first step",
    },
    {
        title: "Step 2",
        content: "This is the second step",
    },
    {
        title: "Step 3",
        content: "This is the third step",
    },
];
it("renders correctly", () => {
    const tree = renderer
        .create(<Stepper steps={steps} currentStep={0} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it("renders correctly", () => {
    const { getByText } = render(<Stepper steps={steps} currentStep={0} />);
    if (steps.length > 1) {
        const nextButton = getByText("Next");
        expect(nextButton).toBeInTheDocument();
    }
});

it("renders correctly", () => {
    const { getByText } = render(<Stepper steps={steps} currentStep={1} />);
    const prevButton = getByText("Prev");
    expect(prevButton).toBeInTheDocument();
});
it("renders correctly", () => {
    const { getByText } = render(<Stepper steps={steps} currentStep={0} />);
    const nextButton = getByText("Next");
    expect(nextButton).toBeInTheDocument();
    fireEvent.click(nextButton);
    expect(getByText("Step 2")).toBeInTheDocument();
});
it("renders correctly", () => {
    const { getByText } = render(<Stepper steps={steps} currentStep={1} />);
    const prevButton = getByText("Prev");
    expect(prevButton).toBeInTheDocument();
    fireEvent.click(prevButton);
    expect(getByText("Step 1")).toBeInTheDocument();
});
it("renders correctly", () => {
    const { queryByText } = render(<Stepper steps={steps} currentStep={2} />);
    const nextButton = queryByText("Next");
    expect(nextButton).not.toBeInTheDocument();
});

it("renders correctly", () => {
    const { queryByText } = render(<Stepper steps={steps} currentStep={0} />);
    const prevButton = queryByText("Prev");
    expect(prevButton).not.toBeInTheDocument();
});
