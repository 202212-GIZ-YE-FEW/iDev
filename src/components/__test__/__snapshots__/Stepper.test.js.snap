// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex items-start justify-between sm:p-6 lg:p-8 flex-col"
>
  <div
    className="max-w-sm p-6 border-gray rounded-lg shadow dark:border-gray"
  >
    <div
      className="block w-full h-full p-4 text-center"
    >
      <p
        className="text-3xl font-normal text-start"
      >
        Step 1
      </p>
      This is the first step
    </div>
    <div
      className="hidden w-full h-full p-4 text-center"
    >
      <p
        className="text-3xl font-normal text-start"
      >
        Step 2
      </p>
      This is the second step
    </div>
    <div
      className="hidden w-full h-full p-4 text-center"
    >
      <p
        className="text-3xl font-normal text-start"
      >
        Step 3
      </p>
      This is the third step
    </div>
    <div
      className="flex flex-row gap-4"
    >
      <button
        className="font-normal cursor-pointer capitalize text-xl rounded-md px-[40px] py-[8px] bg-cyan text-black"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </div>
</div>
`;
